spec/index.bs

```diff
@@ -1249,17 +1249,9 @@ partial interface mixin WindowOrWorkerGlobalScope {
 </pre>
 
 To the [[HTML5#timer-initialisation-steps|timer initialization steps algorithm]],
-add this step between 7.1 and 7.2:
+change step 8.3.2 as follows:
 
-1.  If the first operation argument is not a {{Function}}, set the first operation argument to the result of executing
-    the [$Get Trusted Type compliant string$] algorithm, with
-    *   |global| set to the [=this=] value's [=relevant global object=].
-    *   |input| set to the first method argument, and
-    *   |expectedType| set to {{TrustedScript}}.
-    *   |sink| set to `Window setInterval` if <var ignore>repeat</var> is true, `Window setTimeout` otherwise.
-    *   |sinkGroup| set to `'script'`.
-
-        Note: This matches the logic that the extended attribute would apply.
+1.  Perform <del>HostEnsureCanCompileStrings</del><ins>EnsureCSPDoesNotBlockStringCompilation</ins>(<var>realm</var>, « », <var>handler</var>,<del> false</del><ins>, <var>handler</var>, ~timer~, « », <var>handler</var></ins>). If this throws an exception, catch it, report the exception, and abort these steps.
```

Comment:
This should link to https://w3c.github.io/webappsec-csp/#can-compile-strings.

Reply:
Oh, that's modified in this patch too.

TBH it would be much clearer if the spec changes were upstreamed and merged in a way where the upstream changes don't break upstream. I.e., couldn't https://pr-preview.s3.amazonaws.com/w3c/trusted-types/481/1eee73a...lukewarlow:2c0de9f.html#csp-eval get upstreamed first?

Reply:
Yeah I'm definitely going to upstream things as soon as possible. It's all waiting on the tc39 lynchpin.